CMakeLists.txt

```diff
@@ -326,108 +326,124 @@ function(determine_repository_version source_dir build_repository have_build_rep
+if(DEFINED GIT_EXE AND IS_DIRECTORY "${CMAKE_SOURCE_DIR}/.git")
   determine_repository_version(${CMAKE_SOURCE_DIR} ARANGODB_BUILD_REPOSITORY HAVE_ARANGODB_BUILD_REPOSITORY)
 else()
   set(ARANGODB_BUILD_REPOSITORY "")
   set(HAVE_ARANGODB_BUILD_REPOSITORY "0")
 endif()
 
-if (DEFINED GIT_EXE AND USE_ENTERPRISE AND IS_DIRECTORY "${CMAKE_SOURCE_DIR}/enterprise/.git")
+if(DEFINED GIT_EXE AND USE_ENTERPRISE AND IS_DIRECTORY "${CMAKE_SOURCE_DIR}/enterprise/.git")
   determine_repository_version(${CMAKE_SOURCE_DIR}/enterprise ENTERPRISE_BUILD_REPOSITORY HAVE_ENTERPRISE_BUILD_REPOSITORY)
-else ()
+else()
   set(ENTERPRISE_BUILD_REPOSITORY "")
   set(HAVE_ENTERPRISE_BUILD_REPOSITORY "0")
 endif()
 
-if (DEFINED OSKAR_BUILD_REPOSITITORY)
+if(DEFINED OSKAR_BUILD_REPOSITITORY)
   set(HAVE_OSKAR_BUILD_REPOSITORY "1")
-else ()
+else()
   set(OSKAR_BUILD_REPOSITORY "")
   set(HAVE_OSKAR_BUILD_REPOSITORY "0")
-endif ()
+endif()
 
 configure_file(
   "${CMAKE_CURRENT_SOURCE_DIR}/lib/Basics/build-repository.h.in"
   "${CMAKE_CURRENT_BINARY_DIR}/lib/Basics/build-repository.h"
   NEWLINE_STYLE UNIX
 )
 
-if (VERBOSE)
+if(VERBOSE)
   message(STATUS "ARANGODB_BUILD_REPOSITORY=\"${ARANGODB_BUILD_REPOSITORY}\"")
   message(STATUS "ENTERPRISE_BUILD_REPOSITORY=\"${ENTERPRISE_BUILD_REPOSITORY}\"")
-endif ()
+endif()
 
 ################################################################################
 ## OPERATING SYSTEM
 ################################################################################
 
 if(CMAKE_SYSTEM_NAME MATCHES ".*Linux")
   set(LINUX TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "kFreeBSD.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES "kFreeBSD.*")
   set(FREEBSD TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "kNetBSD.*|NetBSD.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES "kNetBSD.*|NetBSD.*")
   set(NETBSD TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "kOpenBSD.*|OpenBSD.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES "kOpenBSD.*|OpenBSD.*")
   set(OPENBSD TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES ".*GNU.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES ".*GNU.*")
   set(GNU TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES ".*BSDI.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES ".*BSDI.*")
   set(BSDI TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "DragonFly.*|FreeBSD")
+elseif(CMAKE_SYSTEM_NAME MATCHES "DragonFly.*|FreeBSD")
   set(FREEBSD TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "SYSV5.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES "SYSV5.*")
   set(SYSV5 TRUE)
-endif ()
+endif()
+
+# ------------------------------------------------------------------------------
+# LINUX system call GETTID
+# ------------------------------------------------------------------------------
+
+option(CHECK_GETTID "check for the existance of gettid" ON)
+
+if(CHECK_GETTID)
+  message(STATUS "Checking for gettid, use '-DCHECK_GETTID=Off' if cross compiling")
+  try_compile(GETTID_COMPILE_TEST
+          SOURCE_FROM_FILE "check_tid_compilex.c" ${CMAKE_SOURCE_DIR}/cmake/check_tid_compile.c
+          OUTPUT_VARIABLE GETTID_COMPILE_RESULT
```

Comment:
I think defining the `OUTPUT_VARIABLE` is optional: We don't need it, so I would prefer to remove this line.

Reply:
You are right. It is optional. Will remove